src/state/slices/tradeQuoteSlice/selectors.ts

```diff
@@ -27,18 +28,26 @@ import { selectCryptoMarketData, selectFiatToUsdRate } from '../marketDataSlice/
 
 const selectTradeQuoteSlice = (state: ReduxState) => state.tradeQuoteSlice
 
-// TODO(apotheosis): Cache based on quote ID
-export const selectSelectedQuote: Selector<ReduxState, TradeQuote2 | undefined> =
+export const selectSelectedSwapperName: Selector<ReduxState, SwapperName | undefined> =
+  createSelector(selectTradeQuoteSlice, swappers => swappers.swapperName)
+
+export const selectQuotes: Selector<ReduxState, ApiQuote[] | undefined> =
+  createDeepEqualOutputSelector(selectTradeQuoteSlice, swappers => swappers.quotes)
+
+export const selectSelectedQuoteApiResponse: Selector<ReduxState, ApiQuote | undefined> =
   createDeepEqualOutputSelector(
-    selectTradeQuoteSlice,
-    swappers => swappers.quote as TradeQuote2 | undefined,
+    selectQuotes,
+    selectSelectedSwapperName,
+    (quotes, selectedSwapperName) =>
+      quotes?.find(quote => quote.swapperName === selectedSwapperName),
```

Comment:
quotes should always be defined? `[]` when empty right?

```suggestion
      quotes.find(quote => quote.swapperName === selectedSwapperName),
```

Reply:
True, the `selectQuotes` selector was typed incorrectly. Fixed in https://github.com/shapeshift/web/pull/4847/commits/c74339b12b9b75e9f5930866150318945d88049c.